packages/web/src/components/user-generated-text/UserGeneratedTextV2.tsx

```diff
@@ -92,12 +110,44 @@ const Link = ({
   const isExternalLink = !isAudiusUrl(url)
   const to = isExternalLink ? formatExternalLink(url) : formatAudiusUrl(url)
 
+  const handleClick = useCallback(
+    (e: MouseEvent<HTMLAnchorElement>) => {
+      if (to) {
+        if (unfurledContentObject) {
+          if (unfurledContentObject.track) {
+            other.onClick?.(
+              e,
+              'track',
+              decodeHashId(unfurledContentObject.track.id) ?? 0
+            )
+          } else if (unfurledContentObject.collection) {
+            other.onClick?.(
+              e,
+              'collection',
+              decodeHashId(unfurledContentObject.collection.id) ?? 0
+            )
+          } else if (unfurledContentObject.user) {
+            other.onClick?.(
+              e,
+              'user',
+              decodeHashId(unfurledContentObject.user.id) ?? 0
+            )
+          }
+        } else {
+          other.onClick?.(e, 'other')
```

Comment:
this confused me. should we call this something besides other? unknown? external?

Reply:
it's just the other props. I can destructure onclick, that'll be cleaner

Reply:
no i mean the string `other`!

Reply:
Oh! yeah I can actually make two distinct categories. `internal` and `external`